src/AspNetCoreModuleV2/RequestHandlerLib/PipeOutputManager.cpp

```diff
@@ -32,6 +37,31 @@ PipeOutputManager::StopOutputRedirection()
     fflush(stdout);
     fflush(stderr);
 
+    // Restore the original stdout and stderr handles of the process,
+    // as the application has either finished startup or has exited.
+
+    // If stdout/stderr were not set, we need to set it to NUL:
+    // such that other calls to Console.WriteLine don't use an invalid handle
+    FILE *stream;
+
+    if (m_fdPreviousStdOut >= 0)
+    {
+        LOG_IF_DUPFAIL(_dup2(m_fdPreviousStdOut, _fileno(stdout)));
```

Comment:
those are important failure which may lead to AV. I think we should log to windows event log so that user can immediately identifity the root cause. Otherwise user needs to set debug flag and restart the process which may not repro the failure.

Reply:
How does it differ from hundreds of other HRESULTs we don't log?

Reply:
I think special casing this failure and logging to the windows event log is a bit much, however I do think it is important to capture these startup logs. Proposal here: https://github.com/aspnet/IISIntegration/issues/974

Reply:
for application activation path, if there is a failure blocking activation, we always log it somewhere as it is critical.
for non-critical error, e.g., cannot create the log file given by user's configuration, we log a warning.  
In this case, we swallow the error which could be critical, e.g., reset the stdout failed will lead to AV (this is the motivation of the fix, it is worth to logging it in windows event log.